src/typescript/sdk/src/indexer-v2/types/index.ts

```diff
@@ -51,6 +51,11 @@ const toTransactionMetadata = (
   insertedAt: data.inserted_at ? postgresTimestampToDate(data.inserted_at) : new Date(0),
 });
 
+const toBlockAndEventIndex = (data: BlockAndEventIndexMetadata) => ({
+  blockNumber: BigInt(data.block_number),
+  eventIndex: BigInt(data.event_index),
+});
+
```

Comment:
```suggestion
const toBlockAndEventIndex = (data: BlockAndEventIndexMetadata) => data.block_number ? {
  blockNumber: BigInt(data.block_number),
  eventIndex: BigInt(data.event_index),
} : undefined;

```

You should just be able to have this return `undefined` if `block_number` is undefined, and then you don't have to change anything with the `mini-processor` stuff, since the entire `blockAndEvent` field will be undefined, rather than the individual fields

It will work with `curryToNamedType` as expected

Reply:
Yeah just checked!

```typescript
const toBlockAndEventIndex = (data: BlockAndEventIndexMetadata) => ({
  blockNumber: BigInt(data.block_number),
  eventIndex: BigInt(data.event_index),
});

```
const possiblyWithData = curryToNamedType(
  (argsIn: number) => argsIn === 2 ? { data: argsIn } : undefined,
  "maybeData"
);

const res = possiblyWithData(1);
```

and then typescript resolves the type to: 

```typescript
const res: {
    maybeData: {
        data: number;
    } | undefined;
}
```